File: C++/Find missing in second array.cpp
class Solution{
    public:
    vector<int> findMissing(int a[], int b[], int n, int m) 
    {
        vector<int> res;
        unordered_set <int> s;
        for(int i=0; i<m; i++) {
            s.insert(b[i]);
        }
        for(int i=0; i<n; i++) {
            if(s.find(a[i]) == s.end()) res.push_back(a[i]);
        }
        return res;
    } 
};

File: JAVA/Find missing in second array.java
class Solution
{
    ArrayList<Integer> findMissing(int a[], int b[], int n, int m)
    {
        ArrayList<Integer> res = new ArrayList<>();
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int i = 0; i < m; i++){
            map.put(b[i], 1);
        }

        for(int j = 0; j < n; j++){
            if(map.get(a[j]) == null){
                res.add(a[j]);
            }
        }

        return res;
    }
}
